### ISSUE TITLE
error: invalid conversion

### ISSUE THREAD

**Issue by fleixi** [mer-hybris/qt5-qpa-hwcomposer-plugin#40]
if i try to compile it on QT5.5 i get this error:

qeglfscontext.cpp: In constructor 'QEglFSContext::QEglFSContext(HwComposerContext*, QEglFSPageFlipper*, const QSurfaceFo
rmat&, QPlatformOpenGLContext*, EGLDisplay)':
qeglfscontext.cpp:66:97: error: invalid conversion from 'EGLConfig {aka void*}' to 'void**' [-fpermissive]
                         , QEglFSIntegration::chooseConfig(display, hwc->surfaceFormatFor(format))
                                                                                                 ^
In file included from qeglfscontext.h:46:0,
                 from qeglfscontext.cpp:42:
/usr/local/Qt-5.5.0/include/QtPlatformSupport/5.5.0/QtPlatformSupport/private/qeglplatformcontext_p.h:59:5: note: initia
lizing argument 4 of 'QEGLPlatformContext::QEGLPlatformContext(const QSurfaceFormat&, QPlatformOpenGLContext*, EGLDispla
y, void**, const QVariant&)'
     QEGLPlatformContext(const QSurfaceFormat &format, QPlatformOpenGLContext *share, EGLDisplay display,
     ^
Makefile:2218: recipe for target '.obj/qeglfscontext.o' failed
make: *** [.obj/qeglfscontext.o] Error 1

**Comment by thp** [comment]
Yeah, the plugin uses some Qt internals that need porting from one Qt version to another.

**Comment by fleixi** [comment]
the same happen with qt5.4.2 so what’s the last version working with this plugin?

**Comment by rburchell** [comment]
The only heavily tested version is the Qt version in https://github.com/mer-qt/, which is (very roughly) at a heavily patched 5.2 level, but others have tried other versions. Feel free to submit fixes for newer versions.

**Comment by Herrie82** [comment]
This should most probably be fixed with https://github.com/mer-hybris/qt5-qpa-hwcomposer-plugin/pull/80

We have been using the plugin since early Qt5 releases and are now up to Qt 5.12.3 and it's working with above PR.
